web-marketplace/src/components/organisms/LoginButton/LoginButton.tsx

```diff
@@ -125,10 +129,27 @@ const WalletButton = ({ size = 'small' }: Props) => {
           )}
         </>
       </div>
-      <WalletModal
+      <LoginModal
         open={isModalOpen}
         onClose={onModalClose}
         wallets={walletsUiConfig}
+        socialProviders={socialProviders}
+        onEmailSubmit={async values => {
+          if (token && values.email) {
+            await fetch(`${apiUri}/marketplace/v1/auth/magiclogin`, {
+              method: `POST`,
+              body: JSON.stringify({
+                destination: values.email,
+              }),
+              credentials: 'include',
+              headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json',
+                'X-CSRF-TOKEN': token,
+              },
+            });
```

Comment:
We might want to use a small util to do fetch POST/PUT.
So we don't have to repeat headers and error handling every time.
Something like that:

```
export const postData = async ({ url, data, token, method = "POST" }: Params) => {
  const response = await fetch(url, {
    method,
    headers: new Headers({
      Accept: 'application/json',
       'Content-Type': 'application/json',
       'X-CSRF-TOKEN': token,
    }),
    credentials: "include",
    body: JSON.stringify(data),
  });

  if (!response.status !== 200) {
    console.log("Error in postData", { url, data, res });

    throw Error(res.statusText);
  }

  return response.json();
};
```

Reply:
good point, I'll also display the error banner in case there's an error thrown